RationaleMCP/0031/functions.md

```diff
@@ -0,0 +1,95 @@
+# Built-in functions and operators
+
+## In Flat Modelica
+
+### No discussion needed
+
+* abs
+* sign
+* sqrt
+* div
+* mod
+* rem
+* ceil
+* floor
+* integer
+* sin
+* cos
+* tan
+* asin
+* acos
+* atan
+* atan2
+* sinh
+* cosh
+* tanh
+* exp
+* log
+* log10
+* initial
+* terminal
+
+### Probably should be in Flat Modelica as is
+
+* ndims
+* delay
+* spatialDistribution
+* homotopy
+* sample
+* size(A,i)
+* size(A)
+* scalar
+* vector
+* matrix
+* cat
+
+### Odd ones
+
+* String (quite complicated function with many kinds of arguments; could be de-overloaded before Flat Modelica)
+* promote (a built-in operator used to describe other operators)
+
+### Do we need restrictions compared to Modelica?
+
+* der
+* noEvent
+* smooth
+* pre
+* edge
+* change
+* reinit
+
+### Could be implemented in Flat Modelica
+
+Generated function names might need to be changed for different types, etc.
+
+* EnumType(i) (indexed constant array, etc)
+* Integer(enum) (easy to implement even if it could be a Flat Modelica function)
+* semiLinear
```

Comment:
I don't think it's possible to implement `semiLinear` in Flat Modelica, as the semantics depends on being able to identify certain forms of equations involving `semiLinear`.

Reply:
Web Meeting:
Should semiLinear be a built-in function:
Yes: 4
No: 0
Apply this change [Martin]

Reply:
Should EnumType be a built-in function?
Yes: 1
No: 1
To be decided later